### ISSUE TITLE
Error: Unit tests have been faild on Travis

### ISSUE THREAD

**Issue by horike37** [serverless/serverless#4350]
# This is a Bug Report

## Description

For bug reports:
* What went wrong?

Unit tests have been failed since a couple of previous commits.
https://travis-ci.org/serverless/serverless

The infinite loop has occurred within the following test due to the recursive call if 429 error occurs.
https://github.com/serverless/serverless/blob/master/lib/plugins/aws/provider/awsProvider.test.js#L155

We should fix this ASAP so that we can merge PRs
cc: @serverless/framework-maintainers 

Similar or dependent issues:
*maybe #4139

**Comment by horike37** [comment]
I investigated that in the morning, but still I'm not sure.
if comment out the following lines, works fine, but I guess this is not a correct way.
https://github.com/serverless/serverless/blob/master/bin/test#L5-L7

The best way is.. Umm..

**Comment by pmuens** [comment]
Thanks for pointing that out @horike37 👍

We'll look into that...

**Comment by pmuens** [comment]
Hmm. Looks like a recent merge caused those problems. A git bisect could help here.

@horike37 are you able to reproduce this locally?

**Comment by horike37** [comment]
@pmuens 

>@horike37 are you able to reproduce this locally?

Yup, when running `npm run test` on the top of master, reproduce it.
Here is my local info.
```
  Your Environment Information -----------------------------
     OS:                     darwin
     Node Version:           6.9.2
     Serverless Version:     1.23.0
```

Strictly speaking, I was not able to reproduce at first, but re-building the fresh Serverless code by `git clone`, then reproduce at this time.

**Comment by pmuens** [comment]
🤔 interesting! Thanks for confirming @horike37 👍

**Comment by horike37** [comment]
Hey @pmuens 
I continue to investigate that.
I have reverted merge commit one each and tried `npm run test` and eventually backed in the commit of v1.23. 0 release, but still failing.
I guess that there is a minor update of any dependencies package, which caused the problem 🤔 
Then, I have noticed `package-lock.json` was lost. Any ideas why? Is it accidentally?
I'd like to get the past `package-lock.json` or `npm-sharkwrap.json` and test again.
